spec/features/state_file/review_page_spec.rb

```diff
@@ -1,15 +1,20 @@
 require "rails_helper"
 
-RSpec.feature "Completing a state file intake", active_job: true do
+RSpec.feature "Completing a state file intake", active_job: true, js: true do
   include MockTwilio
   include StateFileIntakeHelper
 
+  def wait_for_device_info
+    sleep(2)
+    expect(page.find_all('input[name="state_file_income_review_form[device_id]"]', visible: false).last.value).to be_present
```

Comment:
it looks like the tests pass with just the `sleep(2)` or (`sleep(1)` for me locally). I'm guessing the next `expect` is to just make sure we can still get to the next page correctly (and documentation purposes), which seems good.

I wonder if this is going to make our tests run slightly longer on circle ci? Is there any way to slow this down only on dev and not on circle ci?

also, where is the device info javascript that runs?

Reply:
Thanks for poking on this. Kevin had actually linked me to a [blog post](https://nts.strzibny.name/avoid-sleep-rails-system-tests/) about sleeping in Capybara tests which includes a method for waiting for something in a tight loop with some timeout, and shows how to use Capybara's default timeout for the purpose. I switched over to that, so it should only add as much time is needed for the element to show up, which seems like it should be no additional time on CircleCI (given that these were already passing there).

The device info javascript is in app/javascript/lib/efile_security_information.js, and is called on app/javascript/listeners/index.js:57